prober/tcp_test.go

```diff
@@ -66,6 +67,113 @@ func TestTCPConnectionFails(t *testing.T) {
 	}
 }
 
+func TestTCPConnectionWithTLS(t *testing.T) {
+	ln, err := net.Listen("tcp", ":0")
+	if err != nil {
+		t.Fatalf("Error listening on socket: %s", err)
+	}
+	defer ln.Close()
+	_, listenPort, _ := net.SplitHostPort(ln.Addr().String())
+
+	testCTX, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+
+	// Create test certificates valid for 1 day.
+	certExpiry := time.Now().AddDate(0, 0, 1)
+	testcert_pem, testkey_pem := generateTestCertificate(certExpiry, false)
+
+	// CAFile must be passed via filesystem, use a tempfile.
+	tmpCaFile, err := ioutil.TempFile("", "cafile.pem")
+	if err != nil {
+		panic(fmt.Sprintf("Error creating CA tempfile: %s", err))
```

Comment:
You can t.Fatalf here

Reply:
I decided for panic() because it is not related to the specific test but indicates some lower-level problem. Same panic() went through your last review in `TestTCPConnectionQueryResponseStartTLS`.
For consistency we should either change both or none. I have no strong opinion here and will change both places to t.Fatalf.